### PR TITLE
[SPARK-45222][PYTHON][DOCS] Refine docstring of `DataFrameReader.json`

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to improve the docstring of `DataFrameReader.json`.

### Why are the changes needed?

For end users, and better usability of PySpark.

### Does this PR introduce _any_ user-facing change?

Yes, it fixes the user facing documentation.

### How was this patch tested?

Manually tested.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -380,22 +380,72 @@ def json(
...
+        | 25|Alice|
+        | 30|  Bob|
+        +---+-----+
+
+        Example 3: Read JSON from an RDD of JSON strings
+
+        >>> json_strings = ["{'name': 'Alice', 'age': 25}", "{'name': 'Bob', 'age': 30}"]
+        >>> rdd = spark.sparkContext.parallelize(json_strings)  # doctest: +SKIP
+        >>> df = spark.read.json(rdd)  # doctest: +SKIP
+        >>> df.show()  # doctest: +SKIP
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:426]: Skipped because it doesn't work with Spark Connect, and the result might be indeterministic itself.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by HyukjinKwon** [comment]
Thank you @dongjoon-hyun !!!

@allisonwang-db BTW do you plan to do this for all other functions, or some frequently used only?
With my PRs, (almost) all under SPARK-44728 are resolved. Let's get the rest done, or file JIRAs for all (if that's your intention)

File: python/pyspark/sql/readwriter.py (hunk 1 above)
Comment (allisonwang-db) [reply python/pyspark/sql/readwriter.py:426]: Shall we remove or change this RDD example to something that can work with Spark Connect?

**Review by allisonwang-db — APPROVED** [review]
@HyukjinKwon Thanks for working on this! I only created JIRA for a couple of frequently used ones, but I think we should continue to improve this whenever we see confusing documentation :)

File: python/pyspark/sql/readwriter.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/readwriter.py:426]: Actually, Spark Connect doesn't have a way to convert DataFrame or RDD. So there's no way of doing similar stuff with Spark Connect.

Comment (allisonwang-db) [reply python/pyspark/sql/readwriter.py:426]: Yea my concern is that if users are using Spark Connect, this example won't work for them...

Comment (HyukjinKwon) [reply python/pyspark/sql/readwriter.py]: okie dokie. removed

**Comment by HyukjinKwon** [comment]
Merged to master.

**Review by dongjoon-hyun — APPROVED** [review]

**Pull request closed without merging**